### DISCUSSION TITLE
How to extract text from rectangles larger than CropBox

### DISCUSSION THREAD

**Discussion by gregoriopellegrino** [pymupdf/PyMuPDF#1603] [Q&A] [answered]
## Describe the bug
I am trying to extract text from a PDF for offset print that in the mediabox (outside the trimbox) contains text for printing worflow. I would like this extra-page information not to be extracted with the get_text, so I've set the clip argument equal to the bleed box, but the method always returns text that is outside the page as well.

## To Reproduce
Sample file: [p5.pdf](https://github.com/pymupdf/PyMuPDF/files/8102594/p5.pdf)

```
pdf_file = fitz.open("p5.pdf")
page = pdf_file.load_page(0)
print(page.get_text("text", clip=page.bleedbox))
```

## Expected behavior
Extract only visible text.

## Screenshots
<img width="970" alt="image" src="https://user-images.githubusercontent.com/5065125/154809673-f9542d51-a425-491b-b0d4-e513b9c17af3.png">

## Your configuration
 - macOS 12.1, 64bit
 - Python 3.9.7, 64bit
 - PyMuPDF 1.19.5, 64bit

**Comment by JorjMcKie** [comment]
Both rectangles cropbox, bleedbox contain all the text!

**Comment by JorjMcKie** [comment]
To confirm draw the rectangles, set cropbox to mediabox and save to a new file.

**Comment by JorjMcKie** [comment, marked as answer]
No - revoking the previous! Forget it.
Text extraction does not / cannot "know" that you are handing in the `/BleedBox`. It has no choice but to assume that clip is a rectangle relative to the cropbox - it will **_not_** be regarded as being relative to the mediabox.
So what always works is choosing the cropbox as clip.
If you want a rectangle **_larger_** than that, you must
* either first enlarge the cropbox to that value and then use cropbox as clip again,
* or set cropbox to mediabox, and then use any rectangle you want as the clip
